src/commons/TestFilesUtils.ts

```diff
@@ -0,0 +1,92 @@
+import { TestUtils } from './TestUtils';
+import path from 'path';
+import fs, { Stats } from 'fs';
+
+/**
+ * Manage test files
+ */
+export namespace TestFilesUtils {
+  /**
+   * Extract testIds from test files
+   * @param files array of test files name
+   * @return list of testIds
+   */
+  export function extractTestIdFromFiles(files: Set<string>): Set<string> {
+    const idsSet: Set<string> = new Set<string>();
+
+    for (const file of files) {
+      idsSet.add(TestUtils.extractNumbersFromString(file));
+    }
+    return idsSet;
+  }
+
+  /**
+   * Return list of test files
+   * @param files array of files path
+   * @return array of test filenames
+   */
+  export function getTestsFiles(files: Set<string>): Set<string> {
+    const testFiles: Set<string> = new Set<string>();
+
+    for (const file of files) {
+      const fileName = path.basename(file);
+      if (isTestFile(fileName)) {
+        testFiles.add(fileName);
+      }
+    }
+    return testFiles;
+  }
+
+  /**
+   * Extract testIds from specific path
+   * @param folderPath path of folder to get the files from
+   * @return list of testIds
+   */
+  export function getTestIdsFromFolder(folderPath?: string): Set<string> {
+    const filesInPath = getListOfFilesRecursively(!folderPath ? __dirname : folderPath);
+    const testFilesInPath = getTestsFiles(filesInPath);
+
+    return extractTestIdFromFiles(testFilesInPath);
+  }
+}
+
+/**
+ * Return list of all files in folder and sub folder
+ * @param folderPath path of folder to get the files from
+ */
+function getListOfFilesRecursively(folderPath: string): Set<string> {
+  return getListOfFilesRecursivelyHelper(folderPath, new Set<string>());
+}
```

Comment:
why is it needed? can getListOfFilesRecursivelyHelper get set optional instead of creating new function?